mono/mini/mini-posix.c

```diff
@@ -1023,15 +1026,27 @@ dump_native_stacktrace (const char *signal, void *ctx)
 
 			if (!leave) {
 				mono_sigctx_to_monoctx (ctx, &mctx);
+
+				gchar *scratch_space = NULL;
+#if HAVE_SYS_MMAN_H
+				const int max_summary_len = 500000;
+				scratch_space = (gchar *) mmap (NULL, max_summary_len, PROT_READ | PROT_WRITE, MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
```

Comment:
I suggest to use `mono_valloc` instead.

Reply:
mono_valloc has ```	BEGIN_CRITICAL_SECTION;
``` logic. It was mentioned in channel that this function takes a lock. Looking at it now, I think mono_valloc might be okay.